src/Command/CompileCommand.php

```diff
@@ -0,0 +1,135 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\TwigView\Command;
+
+use Cake\Console\Arguments;
+use Cake\Console\BaseCommand;
+use Cake\Console\ConsoleIo;
+use Cake\Console\ConsoleOptionParser;
+use Cake\TwigView\Filesystem\Scanner;
+use Cake\TwigView\View\TwigView;
+use Exception;
+
+class CompileCommand extends BaseCommand
+{
+    /**
+     * @var \Cake\TwigView\View\TwigView
+     */
+    private $twigView;
+
+    /**
+     * @inheritdoc
+     */
+    protected function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        $parser->addArgument('type', [
+            'required' => true,
+            'choices' => ['all', 'file', 'plugin'],
+            'help' => 'The type you want to compile.',
+        ]);
+
+        $parser->addArgument('target', [
+            'required' => false,
+            'help' => 'The file or plugin you want to compile.',
+        ]);
+
+        return $parser;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
+    {
+        $type = $args->getArgumentAt(0);
+        $target = $args->getArgumentAt(1);
+
+        $this->twigView = new TwigView();
+        switch ($type) {
+            case 'all':
+                $this->compileAll($io);
+                break;
+
+            case 'file':
+                if ($target === null) {
+                    $io->error('File name is missing.');
+
+                    return static::CODE_ERROR;
+                }
+
+                $this->compileFile($io, $target);
+                break;
+
+            case 'plugin':
+                if ($target === null) {
+                    $io->error('Plugin name is missing.');
+
+                    return static::CODE_ERROR;
+                }
+
+                $this->compilePlugin($io, $target);
+                break;
+
+            default:
+                return static::CODE_ERROR;
+        }
+
+        return static::CODE_SUCCESS;
```

Comment:
seems this will not used when there is `default:` above

Reply:
maybe a throw exception here then?